lib/proxy/util.rb

```diff
@@ -11,16 +11,21 @@ class CommandTask
     # create new task and spawn new thread logging all the cmd
     # output to the proxy log. only the process' output is connected
     # stderr is redirected to proxy error log, stdout to proxy debug log
-    def initialize(acommand)
+    # command can be either string or array (command + arguments)
+    # input must be string
+    def initialize(acommand, aninput = nil)
       @command = acommand
+      @input = aninput
     end
 
     def start(&ensured_block)
       # run the task in its own thread
-      logger.debug "Starting task: #{@command}"
```

Comment:
This still looks useful.

Reply:
Absolutely, I just moved it few lines down so I could actually use thread ID in the message so it's obvious which thread ID messages belongs to which command.

Reply:
The thing is that you now move it to within the `Thread.new` and `popen3`. If either fails, you don't really know. For example, `popen3` can raise an exception. While the command may not say a huge amount, I'm not sure what the error output is if the command in the thread failed. Those were my considerations for keeping it in. Since it's debug, most users won't see it, but can be useful when things break for a weird reason.

Not going to block on it because it may be a bit of an edge case so I'll leave it up to you.